include/nuttx/spinlock.h

```diff
@@ -107,28 +107,7 @@ typedef struct
  *
  ****************************************************************************/
 
-#if defined(CONFIG_ARCH_HAVE_TESTSET)
 spinlock_t up_testset(volatile FAR spinlock_t *lock);
-#elif !defined(CONFIG_SMP)
```

Comment:
There is a change in the logic here, now this function will be built for SMP targets. Is this really intended?

Reply:
I don't think so.  The condition is `#if defined(CONFIG_SPINLOCK) && !defined(CONFIG_ARCH_HAVE_TESTSET)` and SMP depends on `CONFIG_ARCH_HAVE_TESTSET`.

Reply:
You’re right, thanks for the clarification @Ouss4!